src/include/codegen/interpreter/bytecode_builder.h

```diff
@@ -0,0 +1,461 @@
+//===----------------------------------------------------------------------===//
+//
+//                         Peloton
+//
+// bytecode_builder.h
+//
+// Identification: src/include/codegen/interpreter/bytecode_builder.h
+//
+// Copyright (c) 2015-2018, Carnegie Mellon University Database Group
+//
+//===----------------------------------------------------------------------===//
+
+#pragma once
+
+#include <ffi.h>
+#include <llvm/ADT/PostOrderIterator.h>
+#include <llvm/IR/CFG.h>
+#include <llvm/IR/Instructions.h>
+#include <llvm/Support/raw_ostream.h>
```

Comment:
How much of these do you need in the header, and how much can forward-declared here and included in the implementation?

Reply:
Good hint, I added llvm forward declarations where it was possible.